### PR TITLE
Added Coingecko support to the pricer Oracle...

### PR DESCRIPTION
**Pull request by sfichera**
Based in new properties at config.json file:

pricer.coingecko.enabled
pricer.supportedtokens

The pricer will query prices for the given token on cryptocompare and coingecko. If the token is active in both oracles it will return the average price.

### CODE REVIEW

File: Spook/Oracles/CoinGecko.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                {
+                    root = root[baseticker];
+                    var price = root.GetDecimal(quoteSymbol.ToLower());
+                    return price;
+                }
+
+            }
+            catch (Exception ex)
+            {
+                return 0;
```
Comment (merl111) [inline Spook/Oracles/CoinGecko.cs:55]: Please do not just swallow the exception, either log a meaningful error message and return zero after, or log the exception, at least `e.Message`.

If something fails we want to know why it failed.

File: Spook/Oracles/CoinGecko.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+            if (String.IsNullOrEmpty(baseticker))
+                return 0;
+
+            var url = $"https://api.coingecko.com/api/v3/simple/price?ids={baseticker}&vs_currencies={quoteSymbol}";
+
+            try
+            {
+
+                using (var httpClient = new HttpClient())
```
Comment (merl111) [inline Spook/Oracles/CoinGecko.cs:32]: I think Webclient is better suited, it's more lightweight so costs less to setup if a new object is created each time called.

File: Spook/Oracles/CoinGecko.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                {
+                    json = httpClient.GetStringAsync(new Uri(url)).Result;
```
Comment (merl111) [inline Spook/Oracles/CoinGecko.cs:34]: Make sure the result returned is not null.

File: Spook/Oracles/Pricer.cs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    public static class Pricer
+    {
+        public static decimal GetCoinRate(string baseSymbol, string quoteSymbol, string cryptoCompApiKey, bool cgEnabled, PricerSupportedToken[] supportedTokens)
+        {
+            try
+            {
+                Decimal cGeckoPrice = 0;
+                Decimal cCompare = 0;
+
+                cCompare = CryptoCompareUtils.GetCoinRate(baseSymbol, quoteSymbol, cryptoCompApiKey, supportedTokens);
```
Comment (merl111) [inline Spook/Oracles/Pricer.cs:15]: Nitpicking here, but maybe you call it `cComparePrice`  if the other price is called `cGeckoPrice`

**Review by merl111 — CHANGES_REQUESTED** [review]
Thank's for the PR, very appreciated. Just a few comments above.

**Comment by sfichera** [comment]
Hi @merl111 thanks for your feedback! I've implemented all your requests... 👍

**Pull request merged**